homeassistant/helpers/event.py

```diff
@@ -349,41 +353,323 @@ def remove_listener() -> None:
+        # to track
+        if self._info.exception:
+            return True
+
+        # There are no entities in the template
+        # to track so this template will
+        # re-render on EVERY state change
+        if not self._info.domains and not self._info.entities:
+            return True
+
+        return False
+
+    @callback
+    def _create_listeners(self) -> None:
+        if self._info.is_static:
+            return
+
+        if self._needs_all_listener:
+            self._setup_all_listener()
+            return
+
+        if self._info.domains:
+            self._setup_domains_listener()
+
+        if self._info.entities or self._info.domains:
+            self._setup_entities_listener()
+
+    @callback
+    def _cancel_domains_listener(self) -> None:
+        if self._domains_listener is None:
+            return
+        self._domains_listener()
+        self._domains_listener = None
+
+    @callback
+    def _cancel_entities_listener(self) -> None:
+        if self._entities_listener is None:
+            return
+        self._entities_listener()
+        self._entities_listener = None
+
+    @callback
+    def _cancel_all_listener(self) -> None:
+        if self._all_listener is None:
+            return
+        self._all_listener()
+        self._all_listener = None
+
+    @callback
+    def _update_listeners(self) -> None:
+        if self._needs_all_listener:
+            if self._all_listener:
+                return
+            self._cancel_domains_listener()
+            self._cancel_entities_listener()
+            self._setup_all_listener()
+            return
+
+        had_all_listner = self._all_listener is not None
+        if had_all_listner:
+            self._cancel_all_listener()
+
+        domains_changed = self._info.domains != self._last_info.domains
+        if had_all_listner or domains_changed:
+            domains_changed = True
+            self._cancel_domains_listener()
+            self._setup_domains_listener()
+
+        if (
+            had_all_listner
+            or domains_changed
+            or self._info.entities != self._last_info.entities
+        ):
+            self._cancel_entities_listener()
+            self._setup_entities_listener()
+
+    @callback
+    def _setup_entities_listener(self) -> None:
+        entities = set(self._info.entities)
+        for entity_id in self.hass.states.async_entity_ids(self._info.domains):
+            entities.add(entity_id)
+        self._entities_listener = async_track_state_change_event(
+            self.hass, entities, self._refresh
+        )
+
+    @callback
+    def _setup_domains_listener(self) -> None:
+        self._domains_listener = async_track_state_added_domain(
+            self.hass, self._info.domains, self._refresh
+        )
+
+    @callback
+    def _setup_all_listener(self) -> None:
+        self._all_listener = self.hass.bus.async_listen(
+            EVENT_STATE_CHANGED, self._refresh
```

Comment:
Should we throttle this to max once every second ? This can get really intentse

Reply:
I tried to find a real world case where this is going to be an issue but didn't come up with anything.  All the ones I have change listeners rarely.